File: app/src/main/java/com/moviereel/moviereel/movienowplaying/MovieNPPresenter.java
package com.moviereel.moviereel.movienowplaying;

import com.moviereel.moviereel.models.MovieModel;

import java.util.List;

/**
 * MovieReel
 * com.moviereel.moviereel.movienowplaying
 * Created by lusinabrian on 26/10/16.
 * Description: Presenter interface that acts as the middleman between the {@link MovieNowPlaying}
 * and Model layer
 */

interface MovieNPPresenter {
    /**Handles what will happen when the Fragment is resumed*/
    void onResume();

    /**picks the data of the item clicked in the RecyclerView
    start activity for the clicked movie item*/
    void onItemClicked(String bundleKey, List<MovieModel> movieList);

    /**Handles what will happen when the Fragment is destroyed*/
    void onDestroy();
}

File: app/src/main/kotlin/com/moviereel/ui/entertain/movie/nowplaying/MovieNPPresenter.kt
package com.moviereel.ui.entertain.movie.nowplaying

import com.moviereel.data.db.entities.movie.MovieNPEntity
import com.moviereel.di.scopes.PerActivity
import com.moviereel.ui.entertain.base.EntertainPageBasePresenter

/**
 * Created by lusinabrian on 26/10/16.
 * Description: Presenter interface that acts as the middleman between the [MovieNPFragment]
 * and Model layer
 */

@PerActivity
interface MovieNPPresenter<V : MovieNPView> : EntertainPageBasePresenter<V> {

    /**picks the data of the item clicked in the RecyclerView
     * start activity for the clicked movie item */
    fun onItemClicked(bundleKey: String, movieList: List<MovieNPEntity>)

}
